nailgun/entities.py

```diff
@@ -4540,6 +4540,45 @@ def create_missing(self):
             self.prior = results[0]
 
 
+class HTTPProxy(
+        Entity,
+        EntityCreateMixin,
+        EntityDeleteMixin,
+        EntityReadMixin,
+        EntitySearchMixin,
+        EntityUpdateMixin):
+    """A representation of a HTTP Proxy entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'name': entity_fields.StringField(
+                required=True,
+                str_type='alpha',
+                length=(6, 12),
+                unique=True
+            ),
+            'url': entity_fields.URLField(required=True),
+            'username': entity_fields.StringField(),
+            'password': entity_fields.StringField(),
+            'organization': entity_fields.OneToManyField(Organization),
+            'location': entity_fields.OneToManyField(Location),
+        }
+        self._meta = {'api_path': 'api/v2/http_proxies', 'server_modes': ('sat')}
+        super(HTTPProxy, self).__init__(server_config, **kwargs)
+
+    def read(self, entity=None, attrs=None, ignore=None, params=None):
+        """Make sure, password, organization and location is in the ignore list for read.
```

Comment:
why do you want to ignore org/loc ?

Reply:
Because of  [BZ177964](https://bugzilla.redhat.com/show_bug.cgi?id=1779642)